### DISCUSSION TITLE
bind and GeoIP

### DISCUSSION THREAD

**Discussion by iptablesf** [coredns/coredns#6458] [Q&A] [answered]
Hi everybody,, we are using simple bind on ubuntu for our dns server... based on user needs we want to add geoip functionality to our dns server .. we got about 2000 dns zones that serve on one standalone ubuntu server. we write API in php that have CRUD functionality in bind zones and records.
i did not find any document that integrate bind and geoip,, i mean, coredns check if request origin is US return specific IP address and default read bind zone... it seems powerdns has some feature like this and have separated yaml file for geoip settings of domain.. but it's written in c++ and we like to use coredns cause its community and base programming language and easier development .
I would be very happy if someone can send me the corefile configuration file that has bind and automatically reloads the zones and has the geoip and settings.
currently we got simple corefile that listen on 53 and check specific paths for bind zone files:
```
.:53 {
    reload 2s
    auto {
        directory /root/zones
        reload 1s
    }
    transfer {
        to *
        to 127.0.0.1
    }
    log
    errors
}
```

**Comment by SuperQ** [comment, marked as answer]
Are you talking about using GeoIP lookups to return different result views based on a match?

Did you look at the examples in the [geoip plugin documentation](https://coredns.io/plugins/geoip/#examples)?

**Reply by iptablesf** [reply]
thank you for your response my friend.. yes i found that ,, but if i want to check 10 countries for every domain name , my Corefile include more than thousand lines and it's not very clean... can coredns provide a way like geodns? i mean parsing json file as geo settings.
also powerdns do something like this and have yaml file for geoip settings...
is this possible in coredns?

**Reply by SuperQ** [reply]
Err, that's what the Corefile does. Moivng things to json or yaml is just deckchairs.

You can use something like Ansible or other templating language to manage your Corefile.

**Reply by iptablesf** [reply]
can u please provide simple example for 2 domains ?
thanks
